EIPS/eip-3534.md

```diff
@@ -0,0 +1,318 @@
+
+The EIP-2718 `ReceiptPayload` for this transaction is `rlp([status, cumulativeGasUsed, logsBloom, logs])`.
+
+### Definitions
+
+- `chainContext`. The transaction is only valid for blockchain data satisfying ALL OF the annotations.
+- `ANNOTATION_COMPOSITE_PREFIX`. A positive integer between `1` and `0xff` that represents the set of subclass annotations in the `chainContext` (_ie._ _which_ chain context subclasses should the provided values be applied to). This value should be the sum of the subclass' `ANNOTATION_PREFIX`s.
+- `ANNOTATION_PREFIX`s are defined for Subclasses as octal-derived positive integers, limited to the set `2^0,2^1,2^2,2^3,2^4,2^5,2^6,2^7`.
+
+The `chainContext` value should be of the form `ANNOTATION_COMPOSITE_PREFIX || [{subclass value}...]`, where 
+- `...` means "zero or more of the things to the left," and 
+- `||` denotes the byte/byte-array concatenation operator.
+
+The `chainContext` value should be encoded as `ANNOTATION_COMPOSITE_PREFIX || rlp[{subclass value}...]`.
+
+### Validation
+
+The values defined as subclasses below acts as constraints on transaction validity for specific chain contexts.
+Transactions defining contraints which are not satisfied by their chain context should be rejected as invalid.
+Blocks containing invalid transactions should be rejected as invalid themselves, per the _status quo_.
+
+### Subclass Combination
+
+`chainContext` values annotating more than one subclass reference should provide those values in the following sequential order:
+
+1. `ANCESTOR_ID`
+2. `ELIGIBLE_MINER_LIST`
+3. `INELIGIBLE_MINER_LIST`
+4. `EXPIRY`
+
+As above, the `ANNOTATION_COMPOSITE_PREFIX` should be the sum of the designated subclass' `ANNOTATION_PREFIX`s.
+### Subclasses
+
+- An `ANNOTATION_PREFIX` value is used to represent each of the available context subclasses.
+
+#### `ancestorId`
+
+- `ANNOTATION_PREFIX` `1`.
+- `ANCESTOR_ID` `bytes`. A byte array between 4 and 12 bytes in length.
+
+The `ANCESTOR_ID` is a reference to a specific block by concatenating the byte representation of a block number and the first 4 bytes of its hash. 
+The block number's should be encoded as a big endian value and should have left-padding 0's removed.
+The block number value may be omitted in case of reference to the genesis block.
+
+The `ANCESTOR_ID` value should be RLP encoded as a byte array for hashing and transmission.
+
+#### `eligibleMinerList`
+
+- `ANNOTATION_PREFIX` `2`.
+- `ELIGIBLE_MINER_LIST` `[address...]`. A list of addresses.
+- `MAX_ELEMENTS` `3`. The maximum number of addresses that can be provided.
+
+The `ELIGIBLE_MINER_LIST` value is an array of valid [address hashes](https://github.com/ethereum/yellowpaper/blob/41c1837f7b1ddcd65f135763c8e965146cd0ab70/Paper.tex#L321) referencing an exclusive, unique set of allowed [`beneficiary`](https://github.com/ethereum/yellowpaper/blob/41c1837f7b1ddcd65f135763c8e965146cd0ab70/Paper.tex#L336) values.
```

Comment:
Avoid external links, including the Ethereum Yellow paper.  It can be assumed that readers of this will know what an Ethereum address is.  Also, you shouldn't call it an address hash, because an `address` is already a hash/truncation of a public key and `address hash` would be a hash of the address which I'm pretty sure is not what you want.  Same with `beneficiary`, it can be assumed that readers of this will know what a `block beneficiary` is, so just call it that (note the addition of the word `block`).